lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -9395,7 +9395,7 @@ namespace Js
             PropertyId propertyId = isNext ? PropertyIds::next : isThrow ? PropertyIds::throw_ : PropertyIds::return_;
             Var prop = JavascriptOperators::GetProperty(iterator, propertyId, scriptContext);
 
-            if (!isNext && prop == iterator->GetLibrary()->GetUndefined())
+            if (!isNext && JavascriptOperators::IsUndefinedOrNull(prop))
```

Comment:
This is because `GetMethod` (which is used to get iterator's `.return`/`.throw`) returns `undefined` either if the property is `null` or if it is `undefined`